colin/cli/colin.py

```diff
@@ -50,6 +50,9 @@ def cli():
 @click.command(name="check",
                context_settings=CONTEXT_SETTINGS)
 @click.argument('target', type=click.STRING)
+@click.option('--supplementary-target', type=click.STRING,
```

Comment:
Internally the input for colin is an image. I wonder how do we discover the specific dockerfile.

Reply:
Oh, is colin not shown the Dockerfile used to build the image?

I wonder if this check is a good fit for colin at all. I saw the 'dockerfile' target and assumed it was something it was also told.

Reply:
Internally, colin testing is initiated when an osbs build completes successfully. The input is only the name of the image (within the registry), colin then fetches it to ostree repo and starts the analysis.

I think it's feasiable to also link to the specific dockerfile which was used to produce the image -- we can still get the dockerfile from inside of the image, right?

Reply:
Only for OSBS-built images. Is that OK?

Reply:
If we can be sure, that OSBS' built images contains the `parent` in metadata, it would be easy.

Reply:
The parent image is indicated in the Koji metadata, not in image labels.

Reply:
Is there some way to add the `parent` to metadata? Otherwise, we need to find and parse the Dockerfile inside of the image.

Reply:
CVP has the glue code between the UMB message and colin: I'm assuming we should be able to change that glue code to also include parent image or a direct link to dist-git to the dockerfile and then just add a new arg to colin to accept the input -- Tim, what you've already implemented.

I agree with Franta here: knowing the parent image will be much efficient -- we won't need to parse any dockerfiles.

Reply:
Sure we can provide any infom from koji task/build.
We just need to agree what info do you need.
* koji_build_id for parent image
* fully qualified name of parent image
* or we can even provide an output of skopeo inspect $fq_name_parent_image)

@lachmanfrantisek @TomasTomecek @twaugh let me know what will be the best for you.

Reply:
Since we still operate on labels, I propose to tell only the fully qualified image name so that colin can perform `skopeo inspect` on its own and do the diff. That should be a simple and efficient implementation.

Reply:
The absolutely ideal input would be the output of `skopeo inspect $fq_name_parent_image`. Then I can just look at `.Labels` and compare directly.

Reply:
I am fine with either way and from unit test POV it might be simpler to have output of `skopeo inspect` But on the other hand could hide network operations.

Choose whatever you/Tomas like more. And I'll provide it in CVP :-)

Reply:
I'm also fine with either way; I'm just slightly concerned with passing json via CLI or env vars, that's it. But since openshift does the same thing and apparently it works fine, let's pass labels directly.

Reply:
Hi, I am not sure how general we want this new check to be.
So we can either pass directly inspect json from parent OR

we can basically pass secondary_target (optional), maybe even secondary_target_type (or we can use same target_type for secondary_target)

that way we can use Target class also for parent target

and then we will just simply use those 2 targets for that inherited optional labels check

what do you think ? @TomasTomecek @lachmanfrantisek @lslebodn

Reply:
just realized that this when we use target_type dockerfile, this check doesn't make sense, so in that case we would either ignore that secondary_target or forbid it with dockerfile type